bindings/python/pydeck/docs/layer.md

```diff
@@ -9,30 +9,171 @@ class pydeck.Layer(
+
+Try changing `type` above to `ScatterplotLayer` and add some `ScatterplotLayer` attributes, `get_fill_color` and `radius`:
+
+```python
+layer = pdk.Layer(
+    'ScatterplotLayer',                 # Change the `type` positional argument here
+    UK_ACCIDENTS_DATA,
+    get_position='[lng, lat]',
+    auto_highlight=True,
+    radius=1000,                        # Radius is given in meters
+    get_fill_color=[180, 0, 200, 140],  # Set an RGBA value for fill
+    pickable=True)
+```
+<img src="https://i.imgur.com/54a7buL.png" alt="pydeck ScatterplotLayer" height=500></img>
+
+## Expression parsers in pydeck layer arguments
+
+One particularly powerful feature of pydeck is an in-built Javascript expression parser that can process a limited subset of Javascript–no functions are allowed, but data accessors, boolean conditions, inline logical statements, arithmetic operations, and arrays are available.
+
+To demonstrate the expression parser, change the color input in `get_fill_color` to a string:
+
+```python
+layer = pdk.Layer(
+    'ScatterplotLayer',
+    UK_ACCIDENTS_DATA,
+    get_position='[lng, lat]',
+    auto_highlight=True,
+    radius=1000,
+    get_fill_color='[180, 0, 200, 140]',
+    pickable=True)
+```
+
+The result of the render will be the same as the last image. The expression parser in deck.gl processes the `get_fill_color` argument of `'[180, 0, 200, 140]'` and converts it to a list of constants.
+
+The expression parser has access to the variables in your data, so these operations can be combined:
+
+```python
+layer = pdk.Layer(
+    'ScatterplotLayer',
+    UK_ACCIDENTS_DATA,
+    get_position='[lng, lat]',
+    auto_highlight=True,
+    radius=1000,
+    get_fill_color='[255, lng > 0 ? 200 * lng : -200 * lng, lng, 140]',
+    pickable=True)
+```
+
+In particular, `get_position` takes `[lng, lat]` in many of these examples. This is deck.gl's expression parser reading the data passed to pydeck and extracting longitude and latitude as a coordinate pair.
+
+Suppose you have a CSV as follows, where the location in the first field in the CSV (here, "coordinates")–
+
+```csv
+coordinates,classification
+"[0.0, 0.0]",A
+"[0.0, 0.0]",A
+"[0.0, 1.0]",B
+"[0.0, 1.0]",C
+```
+
+`get_position` here should be specified as `get_position='coordinates'`
+
+If your coordinates are flattened, you will specify your position as `get_position='[lng,lat]'`
+
+```csv
+lng,lat,classification
+0.0,0.0,A
+0.0,0.0,A
+0.0,1.0,B
+0.0,1.0,C
+```
+
+# Example: Vancouver property values
+
+```python
+import pydeck
+
+DATA_URL = "https://raw.githubusercontent.com/uber-common/deck.gl-data/master/examples/geojson/vancouver-blocks.json"
+LAND_COVER = [[[-123.0, 49.196], [-123.0, 49.324], [-123.306, 49.324], [-123.306, 49.196]]]
+
+INITIAL_VIEW_STATE = pydeck.ViewState(
+  latitude=49.254,
+  longitude=-123.13,
+  zoom=11,
+  max_zoom=16,
+  pitch=45,
+  bearing=0
+)
+
+polygon = pydeck.Layer(  
```

Comment:
This layer is there for the shadow effect. Is there away for pydeck users to specify lighting?

Reply:
There is, using the wrapper around deck.gl's `LightSettings`: https://github.com/uber/deck.gl/blob/master/bindings/python/pydeck/pydeck/bindings/light_settings.py, I can add this to the example and document it as well